src/library_safetimers.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright 2020 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+// Safe timers
+var LibrarySafeTimers = {
+  $SafeTimers__postset: 'Module["requestAnimationFrame"] = function Module_requestAnimationFrame(func) { SafeTimers.requestAnimationFrame(func) };\n',
+  $SafeTimers: {
+    // abort and pause-aware versions TODO: build main loop on top of this?
```

Comment:
Can we take this object's 7 member as make them into top level library members?

That way a program that only needs `safeSetTimeout` would depend only on that one functuion.

Of course each of the function would need to depend on `allowAsyncCallbacks` and `queuedAsyncCallbacks` so maybe that could be done with `autoAddDeps`?

Reply:
Sure. Should they all be prefixed with `emscripten_`?

Reply:
No, these are library internal so should be prefixed with `$`.

Anything without a `$` is visible as a C/C++ and we don't want that in this case I think.

The `$` is only need in the `__deps` and on the decl itself..  When you call the function you don't need the `$`.